### PR TITLE
Add `scale` argument to `Plotter.add_ruler` method

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

resolve #6090

### Details

- None

### CODE REVIEW

File: pyvista/plotting/renderer.py [hunk 1]
```diff
@@ -3870,6 +3874,9 @@ def add_ruler(
         ruler.SetMinorTickLength(minor_tick_length)
         ruler.SetTickOffset(tick_label_offset)
 
+        min_, max_ = ruler.GetRange()
+        ruler.SetRange(min_ * scale, max_ * scale)
```
Comment (MatthewFlamm) [inline pyvista/plotting/renderer.py:3878]: It is redundant to be using `SetRange` twice.  We could set it the first time as 

`ruler.SetRange(0, distance * scale)`  and the inverse.

Comment (tkoyama010) [reply pyvista/plotting/renderer.py:3878]: Oh, it's true! I was about to add bug. Thank you.

Comment (tkoyama010) [reply pyvista/plotting/renderer.py]: resolved in 40526384db1bab4e571dba84f8139e567269d06d.

File: pyvista/plotting/renderer.py [hunk 2]
```diff
@@ -3800,6 +3801,9 @@ def add_ruler(
             Either a string, rgb list, or hex color string for
             tick line colors.
 
+        scale : float, default: 1.0
+            Scale factor for the ruler.
+
```
Comment (tkoyama010) [inline pyvista/plotting/renderer.py:3806]:
```suggestion

            .. versionadded:: 0.44.0
```

File: pyvista/plotting/renderer.py [hunk 3]
```diff
@@ -3800,6 +3801,10 @@ def add_ruler(
...
+            .. versionadded:: 0.44.0
```
Comment (tkoyama010) [inline pyvista/plotting/renderer.py:3807]:
```suggestion
            .. versionadded:: 0.44.0

```

**Comment by tkoyama010** [comment]
@pyvista-bot preview

File: tests/plotting/test_plotter.py [hunk 4]
```diff
@@ -531,3 +531,11 @@ def test_edge_opacity(sphere):
...
     actor = pl.add_mesh(sphere, edge_opacity=edge_opacity)
     assert actor.prop.edge_opacity == edge_opacity
+
+
+def test_add_ruler_scale():
+    plotter = pv.Plotter()
+    ruler = plotter.add_ruler([-0.6, 0.0, 0], [0.6, 0.0, 0], scale=0.5)
+    min_, max_ = ruler.GetRange()
+    assert min_ == 0.0
+    assert max_ == 0.6
```
Comment (MatthewFlamm) [inline tests/plotting/test_plotter.py:541]:
```suggestion
    assert max_ == 0.6
    
    ruler = plotter.add_ruler([-0.6, 0.0, 0], [0.6, 0.0, 0], scale=0.5, flip_range=True)
    min_, max_ = ruler.GetRange()
    assert min_ == 0.6
    assert max_ == 0.0
```

**Review by MatthewFlamm — COMMENTED** [review]
Let's try to get 100% coverage here.

**Review by MatthewFlamm — APPROVED** [review]
LGTM

**Pull request merged**